src/lib.rs

```diff
@@ -1,5 +1,5 @@
 #![warn(missing_docs)]
-#![no_std]
+#![cfg_attr(not(feature = "std"), no_std)]
```

Comment:
I think this would be better if it's in the same style as `alloc`: 
```rust
#![no_std]

...

#[cfg(feature = "alloc")]
extern crate alloc;
#[cfg(feature = "std")]
extern crate std;
```

Reply:
Sounds good.